appscale/tools/appscale_tools.py

```diff
@@ -215,29 +227,193 @@ def add_keypair(cls, options):
 
   @classmethod
   def describe_instances(cls, options):
-    """Queries each node in the currently running AppScale deployment and
-    reports on their status.
+    """
+    Gets cluster stats and prints it nicely.
 
     Args:
       options: A Namespace that has fields for each parameter that can be
         passed in via the command-line interface.
     """
-    login_host = LocalState.get_login_host(options.keyname)
-    login_acc = AppControllerClient(login_host,
-      LocalState.get_secret_key(options.keyname))
+    try:
+      login_host = LocalState.get_login_host(options.keyname)
+      login_acc = AppControllerClient(login_host,
+        LocalState.get_secret_key(options.keyname))
+      all_public_ips = login_acc.get_all_public_ips()
+      cluster_stats = login_acc.get_cluster_stats()
+    except (faultType, AppControllerException, BadConfigurationException):
+      AppScaleLogger.warn("AppScale deployment is probably down")
+      raise
+
+    # Convert cluster stats to useful structures
+    node_stats = {
+      ip: next((n for n in cluster_stats if n["public_ip"] == ip), None)
+      for ip in all_public_ips
+    }
+    apps_dict = next((n["apps"] for n in cluster_stats if n["apps"]), {})
+    apps = [AppInfo(name, app_info) for name, app_info in apps_dict.iteritems()]
+    nodes = [NodeStats(ip, node) for ip, node in node_stats.iteritems() if node]
+    invisible_nodes = [ip for ip, node in node_stats.iteritems() if not node]
+
+    if options.verbose:
+      AppScaleLogger.log("-"*76)
+      cls._print_nodes_info(nodes, invisible_nodes)
+      cls._print_roles_info(nodes)
+    else:
+      AppScaleLogger.log("-"*76)
 
-    for ip in login_acc.get_all_public_ips():
-      acc = AppControllerClient(ip, LocalState.get_secret_key(options.keyname))
-      AppScaleLogger.log("Status of node at {0}:".format(ip))
-      try:
-        AppScaleLogger.log(acc.get_status())
-      except Exception as exception:
-        AppScaleLogger.warn("Unable to contact machine: {0}\n".
-          format(str(exception)))
+    cls._print_cluster_summary(nodes, apps)
+    cls._print_apps(apps)
+    cls._print_status_alerts(nodes, invisible_nodes)
 
-    AppScaleLogger.success("View status information about your AppScale " + \
-      "deployment at http://{0}:{1}/status".format(login_host,
-      RemoteHelper.APP_DASHBOARD_PORT))
+    dashboard = next(
+      (app for app in apps if app.http == RemoteHelper.APP_DASHBOARD_PORT), None
+    )
+    if dashboard and dashboard.appservers > 1:
+      AppScaleLogger.success(
+        "\nView more about your AppScale deployment at http://{}:{}/status"
+        .format(login_host, RemoteHelper.APP_DASHBOARD_PORT)
+      )
+    else:
+      AppScaleLogger.log(
+        "\nAs soon as AppScale Dashboard is started you can visit it at "
```

Comment:
Not necessarily. Sometimes I give up waiting for dashboard and just `Ctrl+C` `appscale up` command, but after that I still want to see trues status message. Additionally, sometimes AppController just can't start Dashboard at all. For example if there is lack of memory, or there is some bug=)